src/components/MessagesList/MessagesList.vue

```diff
@@ -689,7 +694,11 @@ export default {
 		 * @param {boolean} includeLastKnown Include or exclude the last known message in the response
 		 */
 		async getOldMessages(includeLastKnown) {
+			if (this.stopFetchingOldMessages) {
+				return
+			}
 			// Make the request
+			const prevMessagesList = this.messagesList
 			this.loadingOldMessages = true
 			try {
 				await this.$store.dispatch('fetchMessages', {
```

Comment:
we could also get a response here; then:
- check last message in the array `['conversation_created', 'history_cleared'].includes(response.data.ocs.data.at(-1)?.systemMessage)`
- check response status code (200 contains messages. 304 is empty - end of a list)

Reply:
Good point 🦅 , I assumed response code would be 200 for empty list too.